### PR TITLE
Paper: A Python Pipeline for Rapid Application Development (RAD)

### PR DESCRIPTION
**Pull request by sdc50**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

*Editor*: @stargaser

*Reviewers*: @ivan-marroquin

### CODE REVIEW

**Comment by scoobies** [comment]
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

**Comment by mepa** [comment]
@scoobies build paper

**Comment by scoobies** [comment]
building paper...

**Comment by scoobies** [comment]
{"data"=>{"info"=>"Build for paper erdc-paper_christensen_rapid_application_development scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

**Comment by deniederhut** [comment]
@scoobies mark paper

**Comment by deniederhut** [comment]
@scoobies build status

**Comment by scoobies** [comment]
{"erdc-paper_christensen_rapid_application_development"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmp1852yqgg'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: scott_christensen\n*** Warning: /tmp/tmp1852yqgg/publisher/../output/scott_christensen/paper_stats.json does not exist.\nPDFLaTeX error output:\n================================================================================\nThis is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=pdflatex)\n restricted \\write18 enabled.\nentering extended mode\n(./paper.tex\nLaTeX2e <2017-04-15>\nBabel <3.18> and hyphenation patterns for 3 language(s) loaded.\n(./IEEEtran.cls\nDocument Class: IEEEtran 2015/08/26 V1.8b by Michael Shell\n-- See the \"IEEEtran_HOWTO\" manual for usage information.\n-- http://www.michaelshell.org/tex/ieeetran/\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ppl.fd)\n-- Using IEEE Computer Society mode.\n-- Using 8.5in x 11in (letter) paper.\n-- Using PDF output.\n-- This is a 10 point document.\n) (/usr/share/texlive/texmf-dist/tex/latex/cmap/cmap.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ppl.fd)<<t1.cmap>>)\n(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)\n(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))\n(/usr/share/texlive/texmf-dist/tex/latex/caption/caption.sty\n(/usr/share/texlive/texmf-dist/tex/latex/caption/caption3.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty))\n\nPackage caption Warning: Unsupported document class (or package) detected,\n(caption)                usage of the caption package is not recommended.\nSee the caption package documentation for explanation.\n\n) (/usr/share/texlive/texmf-dist/tex/latex/float/float.sty) (./scipy.sty\n(./status.sty (./draftwatermark.sty (./everypage.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/pdftex.def)))\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg))))\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/mathptmx.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/helvet.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/courier.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsmath.sty\nFor additional information on amsmath, use the `?' option.\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amstext.sty\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsgen.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsbsy.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsopn.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amsfonts.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amssymb.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/tools/bm.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))\n(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))\n\nPackage hyperref Warning: Option `pdftex' has already been used,\n(hyperref)                setting the option has no effect on input line 43.\n\n(/usr/share/texlive/texmf-dist/tex/latex/tools/longtable.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/multirow/multirow.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/base/textcomp.sty\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.def\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.dfu)))\n(/usr/share/texlive/texmf-dist/tex/latex/upquote/upquote.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty\nStyle option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix \n<2008/02/07> (tvz)))\nNo file paper.aux.\n(/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ptm.fd)\n-- Lines per column: 61 (exact).\n(/usr/share/texlive/texmf-dist/tex/latex/caption/ltcaption.sty)\n(/usr/share/texlive/texmf-dist/tex/context/base/mkii/supp-pdf.mkii\n[Loading MPS to PDF converter (version 2006.09.02).]\n) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1phv.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ztmcm.fd)<<ot1.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omlztmcm.fd)<<oml.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omsztmcm.fd)<<oms.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/omxztmcm.fd)<<omx.cmap>>\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ptm.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/upzd.fd)\n(/usr/share/texlive/texmf-dist/tex/latex/psnfss/ts1ptm.fd) (./page_numbers.tex)\n[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}]\n\nLaTeX Warning: No positions in optional float specifier.\n               Default added (so using `tbp') on input line 209.\n\n\nLaTeX Warning: File `images/workflow_selection.png' not found on input line 209\n.\n\n\n! Package pdftex.def Error: File `images/workflow_selection.png' not found: usi\nng draft setting.\n\nSee the pdftex.def package documentation for explanation.\nType  H <return>  for immediate help.\n ...                                              \n                                                  \nl.209 ...umnwidth]{images/workflow_selection.png}}\n                                                  \n!  ==> Fatal error occurred, no output PDF file produced!\nTranscript written on paper.log.\n\n================================================================================\n[X] Paper build failed.\nA build error occurred during the self.retrieve_pdf step.", "build_pdf_path"=>"", "build_status"=>"Build failed, no pdf can be found", "build_timestamp"=>"29/05 19:54"}, "status"=>"fail"}}

**Comment by deniederhut** [comment]
@sdc50 looks like you might have forgotten to commit a file? `Warning: File images/workflow_selection.png' not found on input line 209`

**Comment by deniederhut** [comment]
@scoobies mark server fail

**Comment by deniederhut** [comment]
@scoobies assign @stargaser as editor

**Comment by scoobies** [comment]
Assigned! @stargaser is now the editor

**Comment by sdc50** [comment]
> @sdc50 looks like you might have forgotten to commit a file? `Warning: File images/workflow_selection.png' not found on input line 209`

Thank you. Yes I failed to add my images. They should be pushed up now.

**Comment by stargaser** [comment]
@scoobies build status

**Comment by scoobies** [comment]
{"erdc-paper_christensen_rapid_application_development"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpc99f27ve'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: scott_christensen\n*** Warning: /tmp/tmpc99f27ve/publisher/../output/scott_christensen/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/erdc-paper_christensen_rapid_application_development.pdf", "build_status"=>"success", "build_timestamp"=>"31/05 22:21"}, "status"=>"success"}}

**Comment by deniederhut** [comment]
@scoobies mark server success

**Comment by deniederhut** [comment]
@scoobies mark needs review

**Comment by deniederhut** [comment]
@scoobies add @ivan-marroquin to reviewers

**Comment by scoobies** [comment]
@ivan-marroquin added to the reviewers list!

**Comment by ivan-marroquin** [comment]
Hi all,

is there a pdf document ready to download?

Thanks,
Ivan

**Comment by stargaser** [comment]
@ivan-marroquin the paper PDFs can be found on procbuild.scipy.org . The PDF for this paper is [here](https://procbuild.scipy.org/download/erdc-paper_christensen_rapid_application_development). Thank you for reviewing it!

**Comment by deniederhut** [comment]
Hi! We've moved the deadline for first reviews back one week, to June 16th, to give us a little more time to find reviewers for all the papers this year. You can see the updated full timeline at https://github.com/scipy-conference/scipy_proceedings#timeline-for-2022.

**Comment by deniederhut** [comment]
@scoobies add @Bmg-pcl to reviewers

**Comment by scoobies** [comment]
@Bmg-pcl added to the reviewers list!

File: papers/scott_christensen/rapid_application_development.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  Isosurface visualization from a Helios Speed Sweep run. :label:`fig8`
+
+Conclusion
+----------
+
+The RAD pipeline enables both rapid prototyping and rapid progression to production. This enables researchers to quickly communicate and prove ideas and deliver success products on time. RAD has been instrumental for the mission success of several projects at the ERDC.
+
+
+
```
Comment (ivan-marroquin) [inline papers/scott_christensen/rapid_application_development.rst:164]: Dear Authors,

I read with great interest your manuscript. I found it compelling and easy to understand. I only have one suggestion, pay attention to extremely long sentences (see my comments in attached file). I am hoping that you will make your Python library public available.

Kind regards,

Ivan Marroquin
[Reviewed_manuscript_RAD_Python.PDF](https://github.com/scipy-conference/scipy_proceedings/files/8870602/Reviewed_manuscript_RAD_Python.PDF)

**Comment by deniederhut** [comment]
@scoobies mark needs more review

File: papers/scott_christensen/rapid_application_development.rst (hunk 1 above)
Comment (sdc50) [reply papers/scott_christensen/rapid_application_development.rst:164]: Thanks for your comments. I will work to incorporate your suggestions.

**Comment by deniederhut** [comment]
Hi @bmg-pcl! Friendly reminder that the deadline for initial reviews is tomorrow

**Comment by deniederhut** [comment]
@scoobies remove @bmg-pcl from reviewers

**Comment by scoobies** [comment]
@bmg-pcl is not in the reviewers list

**Comment by deniederhut** [comment]
@scoobies mark pending comment

**Comment by deniederhut** [comment]
@scoobies remove @Bmg-pcl from reviewers

**Comment by scoobies** [comment]
@Bmg-pcl removed from the reviewers list!

**Comment by deniederhut** [comment]
Hi @sdc50 ! How are the revisions coming along?

**Comment by sdc50** [comment]
> Hi @sdc50 ! How are the revisions coming along?

Getting there! I'm hoping to have these wrapped up early next week.

**Comment by sdc50** [comment]
@ivan-marroquin Thank you again for reviewing our paper. I believe I've addressed all of the your comments. Please let me know if there everything was addressed satisfactorily.

**Comment by ivan-marroquin** [comment]
Hi @sdc50,

Thanks for letting me know that  you finished with the corrections. I couldn't find the new version of your manuscript. If you can send me a copy to ivan.micha@gmail.com, I will read it again. Otherwise, if you addressed my suggestions. Then, I think the paper is ready for the conference.

Ivan

**Comment by deniederhut** [comment]
@ivan-marroquin you can see the compiled paper at https://procbuild.scipy.org/download/erdc-paper_christensen_rapid_application_development

**Comment by stargaser** [comment]
@scoobies check references

**Comment by scoobies** [comment]
Checking the BibTeX entries failed with the following error: 
```
Couldn't find bibliography entry
```

**Comment by stargaser** [comment]
@scoobies mark ready

**Comment by ivan-marroquin** [comment]
Hi @deniederhut 

Thanks for sharing the link of the revised version of the manuscript. It looks good. Wish you all the best during the conference.

Ivan

**Pull request merged**